lib/chromic_pdf/supervisor.ex

```diff
@@ -125,39 +84,38 @@ defmodule ChromicPDF.Supervisor do
               Supervisor.start_link(children, strategy: :one_for_one, name: MyApp.Supervisor)
             end
         """)
-      end
 
-    result
+    with_browser(supervisor, fn browser ->
+      fun.(%{
+        browser: browser,
+        ghostscript_pool: find_supervisor_child!(supervisor, GhostscriptPool)
+      })
+    end)
   end
 
-  defp try_direct_supervisor(chromic, fun) do
-    name = chromic.get_dynamic_name()
-
-    if pid = Process.whereis(name) do
-      {:found, fun.(pid)}
+  defp with_browser(supervisor, fun) do
+    if agent = find_supervisor_child(supervisor, Agent) do
+      with_on_demand_browser(agent, fun)
     else
-      :not_found
+      supervisor
+      |> find_supervisor_child!(Browser)
+      |> fun.()
     end
```

Comment:
Wouldn't it make more sense to swap these around and rather find the `Browser` first, since that seems the more default use case?